File: frontend/src/constants/storyModels.ts
import type { StoryModelId } from '../types/story';

export const defaultStoryModel: StoryModelId = 'google/gemma-4-26b-a4b-it';

export const storyModels: Array<{ id: StoryModelId; label: string }> = [
  { id: defaultStoryModel, label: 'Gemma' },
  { id: 'yandex/gpt-pro-5.1', label: 'Yandex GPT Pro 5.1' },
  { id: 'qwen/qwen3.6-flash', label: 'Qwen 3.6 Flash' },
];

export function getStoryModelLabel(modelId: StoryModelId) {
  return storyModels.find((model) => model.id === modelId)?.label ?? modelId;
}

File: backend/app/core/story_models.py
from typing import Literal


StoryModelId = Literal[
    "google/gemma-4-26b-a4b-it",
    "yandex/gpt-pro-5.1",
    "qwen/qwen3.6-flash",
]

DEFAULT_STORY_MODEL: StoryModelId = "google/gemma-4-26b-a4b-it"
AVAILABLE_STORY_MODELS: tuple[StoryModelId, ...] = (
    DEFAULT_STORY_MODEL,
    "yandex/gpt-pro-5.1",
    "qwen/qwen3.6-flash",
)
